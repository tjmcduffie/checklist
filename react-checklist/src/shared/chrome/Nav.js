import React from 'react';
import {Link} from 'react-router-dom';

const Nav = () => {
  return (
    <nav>
      <ul>
        <li>
          <Link to="/">Home</Link>
          <Link to="/react-classic">React Classic</Link>
          <Link to="/react-hooks">React Hooks</Link>
        </li>
      </ul>
    </nav>
  );
};

export default Nav;
