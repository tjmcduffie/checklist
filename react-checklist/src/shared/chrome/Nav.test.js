import Nav from './Nav';
import React from 'react';
import {create} from 'react-test-renderer';

import {Link} from 'react-router-dom';

jest.mock('react-router-dom', () => ({
  Link: 'Link'
}));

describe('Nav', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const elem = create(<Nav />);
    elem.unmount();
  });

  it('renders predictably', () => {
    const elem = create(<Nav />);
    expect(elem.toJSON()).toMatchSnapshot();
  });
});
