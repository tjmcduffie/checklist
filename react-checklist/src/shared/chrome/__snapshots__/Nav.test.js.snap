// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders predictably 1`] = `
<nav>
  <ul>
    <li>
      <Link
        to="/"
      >
        Home
      </Link>
      <Link
        to="/react-classic"
      >
        React Classic
      </Link>
      <Link
        to="/react-hooks"
      >
        React Hooks
      </Link>
    </li>
  </ul>
</nav>
`;
